### DISCUSSION TITLE
How to enable DeepPartial<{T}> when using the server preset @eddeee888/gcg-typescript-resolver-files

### DISCUSSION THREAD

**Discussion by xoldd** [dotansimha/graphql-code-generator#9658] [Q&A] [answered]
[In the graphql codegen docs it makes use of plugins field.](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-resolvers#allow-deep-partial-with-utility-types)

How to do the import thing when using [@eddeee888/gcg-typescript-resolver-files](https://github.com/eddeee888/graphql-code-generator-plugins)? Without the import DeepPartial will end up as undefined in the generated types.

**Comment by eddeee888** [comment]
Hi @xoldyckk 👋 

`typescript` and `typescript-resolvers` plugin config can be provided through the [typesPluginConfig](https://github.com/eddeee888/graphql-code-generator-plugins/tree/master/packages/typescript-resolver-files#typespluginsconfig) option:

```ts
// codegen.ts
import type { CodegenConfig } from "@graphql-codegen/cli";
import { defineConfig } from "@eddeee888/gcg-typescript-resolver-files";

const config: CodegenConfig = {
  schema: "**/schema.graphql",
  generates: {
    "src/schema": defineConfig({
      typesPluginsConfig: { // using `defineConfig` means `typesPluginsConfig` is typed for `typescript` and `typescript-resolvers` options
        defaultMapper: "DeepPartial<{T}>", 
        // and any other types plugins config
      },
    }),
  },
};
export default config;
```

Note that most but not all config options are being handled (due to being time-poor). I'm waiting for consumers usage/requests before implementing certain config 🙂.

Let me know if you this worked for you

**Comment by xoldd** [comment]
[Answered here](https://discord.com/channels/625400653321076807/631489012632125440/1148231382913531986)

**Reply by eddeee888** [reply, marked as answer]
There's now official support in v0.7.0: https://github.com/eddeee888/graphql-code-generator-plugins/releases/tag/%40eddeee888%2Fgcg-typescript-resolver-files%400.7.0

Thanks for raising the issue and pinging me 🙌
